python/ray/serve/controller.py

```diff
@@ -519,6 +534,27 @@ async def __init__(self,
 
         asyncio.get_event_loop().create_task(self.run_control_loop())
 
+    async def wait_for_event(self, uuid: ResultId) -> FutureResult:
+        future = self.inflight_results[uuid]
+        result = await future
+        # NOTE(ilr) We have to delete this *after* we return to avoid a
+        # situation where the controller crashes *after* deleting and before
+        # returning to the API (and user).
```

Comment:
Oh man this is kind of ugly.

Reply:
Fine for now, but maybe we can think of a cleaner solution

Reply:
Yeah, it isn't great, but it's correct.